pandas/core/indexing.py

```diff
@@ -1684,6 +1684,13 @@ def _setitem_with_indexer_split_path(self, indexer, value):
                 for loc, v in zip(ilocs, value):
                     self._setitem_single_column(loc, v, pi)
 
+            elif len(ilocs) == 1 and com.is_null_slice(pi) and len(self.obj) == 0:
```

Comment:
hmm this case is handled above no (before the split path)? (unless you changed the code)

Reply:
in the homogeneous-dtype case this goes through the non-split path.  With mixed dtypes we go through split path, and in master fall through to raise a few lines down